tsl/src/continuous_aggs/materialize.c

```diff
@@ -495,12 +522,42 @@ scan_take_invalidation_tuple(TupleInfo *ti, void *data)
 	return SCAN_CONTINUE;
 }
 
+/* use materialization_invalidation_threshold to filter the logs */
+static ScanTupleResult
+scan_del_materialization_invalidation_tuple(TupleInfo *ti, void *data)
+{
+	InvalidationScanState *scan_state = (InvalidationScanState *) data;
+
+	MemoryContext old_ctx = MemoryContextSwitchTo(scan_state->mctx);
+	Form_continuous_aggs_materialization_invalidation_log invalidation_form =
+		((Form_continuous_aggs_materialization_invalidation_log) GETSTRUCT(ti->tuple));
+	Invalidation *invalidation = palloc(sizeof(*invalidation));
+
+	invalidation->lowest_modified_value = invalidation_form->lowest_modified_value;
+	invalidation->greatest_modified_value = invalidation_form->greatest_modified_value;
+
+	Assert(invalidation->lowest_modified_value <= invalidation->greatest_modified_value);
+	/* this will pick up invalidations where the mat threshold lies inside the
+	 * range. so if we had [1Jan, 20Jan] as the invalidation range and our
+	 * threshold is Jan8. This will pick it up
```

Comment:
would be nice to explain why it's ok to throw the other rows out.

Reply:
added comments